File: swift/495_TeemoAttacking.swift
class Solution {
    func findPoisonedDuration(_ timeSeries: [Int], _ duration: Int) -> Int {
        var totalTime: Int = 0, currEndOverlap: Int = 0
        for t in timeSeries {
            totalTime += min(t + duration - currEndOverlap, duration);
            currEndOverlap = t + duration;
        }
        return totalTime
    }
}

File: python/495_TeemoAttacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        total_time, curr_overlap_end = 0, 0
        for t in timeSeries:
            total_time += min(t + duration - curr_overlap_end, duration)
            curr_overlap_end = t + duration
        return total_time
